src/lib.rs

```diff
@@ -87,17 +89,17 @@ impl<
         d6: D6,
         d7: D7,
         delay: D,
-    ) -> HD44780<D, EightBitBus<RS, EN, D0, D1, D2, D3, D4, D5, D6, D7>> {
+    ) -> core::result::Result<HD44780<D, EightBitBus<RS, EN, D0, D1, D2, D3, D4, D5, D6, D7>>, Error> {
```

Comment:
Can we change this to `Result<HD44780<D, EightBitBus<RS, EN, D0, D1, D2, D3, D4, D5, D6, D7>>>` by using the new result type you've defined?